openedx/core/lib/block_structure/manager.py

```diff
@@ -90,39 +92,48 @@ def get_collected(self):
                 from each registered transformer.
         """
         try:
-            block_structure = BlockStructureFactory.create_from_cache(
+            block_structure = BlockStructureFactory.create_from_store(
                 self.root_block_usage_key,
-                self.block_structure_cache
+                self.store,
             )
             BlockStructureTransformers.verify_versions(block_structure)
 
         except (BlockStructureNotFound, TransformerDataIncompatible):
-            block_structure = self.update_collected()
+            if config.is_enabled(config.RAISE_ERROR_WHEN_NOT_FOUND):
+                raise
+            else:
+                block_structure = self._update_collected()
 
         return block_structure
 
-    def update_collected(self):
+    def update_collected_if_needed(self):
```

Comment:
**Note to reviewers:** Now that we keep track of the version of the collected data in SQL, we can easily verify whether the previously collected data is already up-to-date.